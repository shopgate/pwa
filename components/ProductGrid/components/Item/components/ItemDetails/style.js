import { css } from 'glamor';

export const details = css({
  lineHeight: 1.2,
  ':not(:empty)': {
    padding: '12px 0 30px',
  },
});
