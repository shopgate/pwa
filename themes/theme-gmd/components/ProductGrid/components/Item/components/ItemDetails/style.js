import { css } from 'glamor';

export const details = css({
  lineHeight: 1.35,
  ':not(:empty)': {
    padding: '12px 16px',
  },
});

export const title = css({
  fontWeight: '500',
  lineHeight: 1.15,
  marginTop: 1,
  wordBreak: ['keep-all', 'break-word'],
  hyphens: 'auto',
});
