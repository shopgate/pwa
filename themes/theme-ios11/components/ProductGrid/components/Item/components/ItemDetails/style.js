import { css } from 'glamor';

export const details = css({
  lineHeight: 1.2,
  ':not(:empty)': {
    padding: '12px 0 30px',
  },
});

export const quantityHint = css({
  paddingTop: 8,
}).toString();
