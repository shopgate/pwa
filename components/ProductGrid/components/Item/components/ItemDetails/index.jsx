import React from 'react';
import PropTypes from 'prop-types';
import {
  MapPriceHint,
  OrderQuantityHint,
  Swatches,
  getProductRoute,
  AVAILABILITY_STATE_OK,
  AVAILABILITY_STATE_ALERT,
} from '@shopgate/engage/product';
import {
  TextLink, Link, Availability,
} from '@shopgate/engage/components';
import { i18n } from '@shopgate/engage/core';

import ItemName from '../ItemName';
import ItemPrice from '../ItemPrice';
import ShortDescription from '../ShortDescription';
import * as styles from './style';

/**
 * The item details component.
 * @returns {JSX}
 */
const ItemDetails = ({ product, display }) => {
  const {
    id: productId, name = null, stock = null, shortDescription = null,
  } = product;

  if (display && !display.name && !display.price && !display.reviews) {
    return null;
  }

  return (
    <div className={styles.details} tabIndex={-1} role="button">
      <Link
        href={getProductRoute(productId)}
        state={{ title: name }}
      >
        {/*
          This feature is currently in BETA testing.
          It should only be used for approved BETA Client Projects
        */}
        <Swatches productId={productId} />
      </Link>
      <TextLink
        href={getProductRoute(productId)}
        state={{ title: name }}
        className={styles.itemNameLink}
      >
        <ItemName display={display} productId={productId} name={name} />
      </TextLink>

      <Link
        tag="a"
        href={getProductRoute(productId)}
        state={{ title: name }}
        className={styles.propertiesLink}
      >
        <ShortDescription shortDescription={shortDescription} />

        {/*
          This feature is currently in BETA testing.
          It should only be used for approved BETA Client Projects
        */}
        <MapPriceHint productId={productId} />

        {/*
          This feature is currently in BETA testing.
          It should only be used for approved BETA Client Projects
        */}
        <OrderQuantityHint productId={productId} />

        <Availability
          state={!stock || stock.orderable
            ? AVAILABILITY_STATE_OK
            : AVAILABILITY_STATE_ALERT
          }
          text={i18n.text('product.available.not')}
          showWhenAvailable={false}
        />

        <div className={styles.itemPrice}>
          <ItemPrice product={product} display={display} />
        </div>
      </Link>
    </div>
  );
};

ItemDetails.propTypes = {
  product: PropTypes.shape().isRequired,
  display: PropTypes.shape(),
};

ItemDetails.defaultProps = {
  display: null,
};

export default ItemDetails;
