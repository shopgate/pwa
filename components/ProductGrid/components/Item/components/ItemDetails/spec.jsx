import React from 'react';
import { shallow } from 'enzyme';
import ItemDetails from './index';

jest.mock('@shopgate/engage/product', () => ({
  MapPriceHint: () => null,
  OrderQuantityHint: () => null,
  Swatches: () => null,
  getProductRoute: () => '',
  AVAILABILITY_STATE_OK: 'AVAILABILITY_STATE_OK',
  AVAILABILITY_STATE_ALERT: 'AVAILABILITY_STATE_ALERT',
}));
jest.mock('@shopgate/engage/components');
jest.mock('@shopgate/engage/core');
jest.mock('../ItemName');
jest.mock('../ItemPrice');

describe('<ItemDetails />', () => {
  const props = {
    product: {
      id: '1234',
      name: 'Foo',
      price: {},
    },
  };

  const display = {
    name: false,
    price: false,
    reviews: false,
  };

  it('should render with minimal props', () => {
    const wrapper = shallow(<ItemDetails {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should not render with display props set', () => {
    const wrapper = shallow(<ItemDetails {...props} display={display} />);
    expect(wrapper).toBeEmptyRender();
  });
});
