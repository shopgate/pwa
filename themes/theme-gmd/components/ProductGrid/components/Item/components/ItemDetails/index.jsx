import React, { useCallback, useMemo } from 'react';
import PropTypes from 'prop-types';
import {
  MapPriceHint,
  OrderQuantityHint,
  EffectivityDates,
  Swatches,
  getProductRoute,
  AVAILABILITY_STATE_OK,
  AVAILABILITY_STATE_ALERT,
} from '@shopgate/engage/product';
import { Link, Availability } from '@shopgate/engage/components';
import { StockInfoLists } from '@shopgate/engage/locations/components';
import { hasNewServices as checkHasNewServices, i18n } from '@shopgate/engage/core/helpers';
import { useNavigation } from '@shopgate/engage/core/hooks';
import ItemName from '../ItemName';
import ItemPrice from '../ItemPrice';
import ShortDescription from '../ShortDescription';
import * as styles from './style';

/**
 * The item details component.
 * @returns {JSX.Element}
 */
const ItemDetails = ({ product, display }) => {
  const {
    id: productId, name = null, stock = null, shortDescription = null,
  } = product;

  const hasNewServices = useMemo(() => checkHasNewServices(), []);
  const { push } = useNavigation();

  // click events necessary for a11y navigation on Android
  const handleClick = useCallback(() => {
    push({
      pathname: getProductRoute(productId),
    });
  }, [productId, push]);

  const handleKeyDown = useCallback((event) => {
    if (event.key === 'Enter' || event.key === ' ') {
      handleClick();
    }
  }, [handleClick]);

  if (display && !display.name && !display.price && !display.reviews) {
    return null;
  }

  return (
    <Link
      className={`${styles.details} theme__product-grid__item__item-details`}
      tabIndex={0}
      onClick={handleClick}
      onKeyDown={handleKeyDown}
      href={getProductRoute(productId)}
    >
      <div>
        {/*
          This feature is currently in BETA testing.
          It should only be used for approved BETA Client Projects
        */}
        <Swatches productId={productId} />
      </div>
      <div className={styles.itemNameLink}>
        <ItemName display={display} productId={productId} name={name} />
      </div>

      <div className={styles.propertiesLink}>
        <ShortDescription shortDescription={shortDescription} />

        {/*
          This feature is currently in BETA testing.
          It should only be used for approved BETA Client Projects
        */}
        <MapPriceHint productId={productId} />

        {/*
          This feature is currently in BETA testing.
          It should only be used for approved BETA Client Projects
        */}
        <OrderQuantityHint productId={productId} className={styles.quantityHint} />

        {/* This feature is currently in BETA testing.
          It should only be used for approved BETA Client Projects
        */}
        <EffectivityDates productId={productId} />

        { hasNewServices && (
        <>
          <Availability
            state={!stock || stock.orderable
              ? AVAILABILITY_STATE_OK
              : AVAILABILITY_STATE_ALERT
            }
            text={i18n.text('product.available.not')}
            showWhenAvailable={false}
          />
          <StockInfoLists product={product} />
        </>
        )}

        <div className={styles.itemPrice}>
          <ItemPrice product={product} display={display} />
        </div>
      </div>
    </Link>
  );
};

ItemDetails.propTypes = {
  product: PropTypes.shape().isRequired,
  display: PropTypes.shape(),
};

ItemDetails.defaultProps = {
  display: null,
};

export default ItemDetails;
