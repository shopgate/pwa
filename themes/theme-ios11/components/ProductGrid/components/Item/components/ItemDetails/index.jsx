import React, { useCallback, useMemo } from 'react';
import PropTypes from 'prop-types';
import {
  MapPriceHint,
  OrderQuantityHint,
  EffectivityDates,
  Swatches,
  AVAILABILITY_STATE_OK,
  AVAILABILITY_STATE_ALERT,
  getProductRoute,
} from '@shopgate/engage/product';
import { hasNewServices as checkHasNewServices, i18n } from '@shopgate/engage/core/helpers';
import { Availability, Link } from '@shopgate/engage/components';
import { StockInfoLists } from '@shopgate/engage/locations/components';
import { historyPush } from '@shopgate/pwa-common/actions/router';
import { useDispatch } from 'react-redux';
import ItemName from '../ItemName';
import ItemPrice from '../ItemPrice';
import * as styles from './style';

/**
 * The Product Grid Item Detail component.
 * @param {Object} props The component props.
 * @param {Object} props.product The product.
 * @param {Object} props.display The display object.
 * @returns {JSX.Element}
 */
const ItemDetails = ({ product, display }) => {
  const { id: productId, name = null, stock = null } = product;
  const dispatch = useDispatch();

  const hasNewServices = useMemo(() => checkHasNewServices(), []);

  // click events necessary for a11y navigation on Android
  const handleClick = useCallback(() => {
    dispatch(historyPush({
      pathname: getProductRoute(productId),
    }));
  }, [dispatch, productId]);

  const handleKeyDown = useCallback((event) => {
    if (event.key === 'Enter' || event.key === ' ') {
      handleClick();
    }
  }, [handleClick]);

  if (display && !display.name && !display.price && !display.reviews) {
    return null;
  }

  return (
    <Link
      className={`${styles.details} theme__product-grid__item__item-details`}
      role="button"
      onClick={handleClick}
      onKeyDown={handleKeyDown}
      href={getProductRoute(productId)}
    >
      {/*
        This feature is currently in BETA testing.
        It should only be used for approved BETA Client Projects
      */}
      <Swatches productId={productId} />

      <ItemName display={display} productId={productId} name={name} />

      {/*
        This feature is currently in BETA testing.
        It should only be used for approved BETA Client Projects
      */}
      <MapPriceHint productId={productId} />

      {/*
        This feature is currently in BETA testing.
        It should only be used for approved BETA Client Projects
      */}
      <OrderQuantityHint productId={productId} className={styles.quantityHint} />

      {/* This feature is currently in BETA testing.
      It should only be used for approved BETA Client Projects */}
      <EffectivityDates productId={productId} />
      { hasNewServices && (
        <>
          <Availability
            state={!stock || stock.orderable
              ? AVAILABILITY_STATE_OK
              : AVAILABILITY_STATE_ALERT
        }
            text={i18n.text('product.available.not')}
            showWhenAvailable={false}
          />
          <StockInfoLists product={product} />
        </>
      )}

      <ItemPrice product={product} display={display} />
    </Link>
  );
};

ItemDetails.propTypes = {
  product: PropTypes.shape().isRequired,
  display: PropTypes.shape(),
};

ItemDetails.defaultProps = {
  display: null,
};

export default ItemDetails;
