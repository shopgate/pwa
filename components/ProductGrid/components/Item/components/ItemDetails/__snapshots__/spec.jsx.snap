// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ItemDetails /> should render with minimal props 1`] = `
<div
  className={
    Object {
      "data-css-sr2jsa": "",
    }
  }
  role="button"
  tabIndex={-1}
>
  <Link
    href=""
    state={
      Object {
        "title": "Foo",
      }
    }
  >
    <Swatches
      productId="1234"
    />
  </Link>
  <TextLink
    className="css-nfstwl"
    href=""
    state={
      Object {
        "title": "Foo",
      }
    }
  >
    <ItemName
      display={null}
      name="Foo"
      productId="1234"
    />
  </TextLink>
  <Link
    className="css-1mm725o"
    href=""
    state={
      Object {
        "title": "Foo",
      }
    }
    tag="a"
  >
    <ShortDescription
      shortDescription={null}
    />
    <MapPriceHint
      productId="1234"
    />
    <OrderQuantityHint
      productId="1234"
    />
    <Availability
      showWhenAvailable={false}
      state="AVAILABILITY_STATE_OK"
      text="product.available.not"
    />
    <div
      className="css-193sdzn"
    >
      <ItemPrice
        display={null}
        product={
          Object {
            "id": "1234",
            "name": "Foo",
            "price": Object {},
          }
        }
      />
    </div>
  </Link>
</div>
`;
